src/services/messengerSync/consumer.js

```diff
@@ -8,29 +9,58 @@ const QUEUE_EXTERNAL = Configuration.get('RABBITMQ_MATRIX_QUEUE_EXTERNAL');
 
 let channel;
 
-const validateMessage = (content) => (content.userId && (content.courses || content.teams || content.fullSync));
+const validateMessage = (content) => {
+	if (!content.action) {
+		return false;
+	}
+
+	if (!content.userId && !content.schoolId) {
+		return false;
+	}
+
+	return (content.courses || content.teams || content.fullSync);
+};
+
+const sendToQueue = (message) => {
+	const msgJson = JSON.stringify(message);
+	const msgBuffer = Buffer.from(msgJson);
+	channel.sendToQueue(QUEUE_EXTERNAL, msgBuffer, { persistent: true });
+};
 
 const handleMessage = async (incomingMessage) => {
 	const content = JSON.parse(incomingMessage.content.toString());
 	if (!validateMessage(content)) {
-		logger.warning(`MESSENGER SYNC: invalid message in queue ${QUEUE_INTERNAL}`, incomingMessage);
+		logger.warning(`MESSENGER SYNC: invalid message in queue ${QUEUE_INTERNAL}`, content);
 		// message is invalid an can not be retried
-		channel.reject(incomingMessage, false);
+		return channel.reject(incomingMessage, false);
 	}
 	try {
 		if (!await messengerActivatedForSchool(content)) {
 			// school should not be synced
-			channel.reject(incomingMessage, false);
+			return channel.reject(incomingMessage, false);
 		}
 
-		const addUserMessageObject = await buildAddUserMessage(content);
-		const outgoingMessage = JSON.stringify(addUserMessageObject);
-		channel.sendToQueue(QUEUE_EXTERNAL, Buffer.from(outgoingMessage), { persistent: true });
-		channel.ack(incomingMessage);
+		switch (content.action) {
+			case 'syncSchool': {
+				await requestSyncForEachSchoolUser(content.schoolId);
+				return channel.ack(incomingMessage);
+			}
+
+			case 'syncUser': {
+				const outgoingMessage = await buildAddUserMessage(content);
+				sendToQueue(outgoingMessage);
+				return channel.ack(incomingMessage);
+			}
+
+			default: {
+				// message can't be processed
+				return channel.reject(incomingMessage, false);
+			}
+		}
 	} catch (err) {
 		logger.error(`MESSENGER SYNC: error while handling message in queue ${QUEUE_INTERNAL} `, err);
 		// retry message once (the second time it is redelivered)
-		channel.reject(incomingMessage, !incomingMessage.fields.redelivered);
+		return channel.reject(incomingMessage, !incomingMessage.fields.redelivered);
```

Comment:
this will fail again like above. to retry, use a retry around  handleMessage from its caller like https://tech.mybuilder.com/handling-retries-and-back-off-attempts-with-javascript-promises/